compliance_tool/cli.py

```diff
@@ -1,57 +1,67 @@
 import argparse
-from compliance_tool import list, scan, remediate
+from compliance_tool import list_, scan, remediate
 
 
-def prepare_parser_list(subparsers) -> None:
-    parser = subparsers.add_parser("list",
-                                   help="list available components")
-    parser.set_defaults(func=list.execute)
+def prepare_parser_list(cli_sub_parsers, global_parser, profile_parser) -> None:
+    parser = cli_sub_parsers.add_parser("list", parents=[global_parser],
+                                        help="list available components")
+    parser.set_defaults(func=list_.execute)
     list_subparsers = parser.add_subparsers(required=True)
 
     list_subparsers.add_parser("profiles",
                                help="list available profiles")
 
-    parser = list_subparsers.add_parser("controls",
-                                        help="list available controls for a given profile")
-    parser.add_argument("--profile")
+    list_subparsers.add_parser("controls", parents=[profile_parser],
+                               help="list available controls for a given profile")
 
-    parser = list_subparsers.add_parser("rules",
-                                        help="list available rules for a given profile")
-    parser.add_argument("--profile")
+    list_subparsers.add_parser("rules", parents=[profile_parser],
+                               help="list available rules for a given profile")
 
 
-def prepare_parser_scan(subparsers) -> None:
-    parser = subparsers.add_parser("scan",
-                                   help="perform a compliance scan")
+def prepare_parser_scan(cli_sub_parsers, global_parser, profile_parser) -> None:
```

Comment:
For an unknown reason, I'm able to run `python3 -m compliance_tool.cli scan` where I haven't provided `--profile` and it hasn't produce any error despite it must produce an error. Please take a look into this problem.

Reply:
I agree that the `scan`  and `remediate` verbs should require a profile.

Reply:
What about `list rules`? Shouldn't it require `--profile` as well?

Reply:
That seems reasonable. Listing all rules in the data stream doesn't seem too useful.

Reply:
I agree with Matthew